AdyenDropIn/Utilities/ComponentManager.swift

```diff
@@ -34,26 +36,30 @@ internal final class ComponentManager {
                   context: AdyenContext,
                   configuration: DropInComponent.Configuration,
                   partialPaymentEnabled: Bool = true,
-                  remainingAmount: Amount? = nil,
                   order: PartialPaymentOrder?,
                   supportsEditingStoredPaymentMethods: Bool = false,
                   presentationDelegate: PresentationDelegate) {
         self.paymentMethods = paymentMethods
-        self.context = context
         self.configuration = configuration
         self.partialPaymentEnabled = partialPaymentEnabled
-        self.remainingAmount = remainingAmount
         self.order = order
         self.supportsEditingStoredPaymentMethods = supportsEditingStoredPaymentMethods
         self.presentationDelegate = presentationDelegate
+
+        if let payment = context.payment, let remainingAmount = order?.remainingAmount {
+            let payment = Payment(amount: remainingAmount, countryCode: payment.countryCode)
+            self.context = context.updated(with: payment)
+        } else {
+            self.context = context
+        }
```

Comment:
I think there is a potential issue here, that the ComponentManager. context will be different than DropInComponent.context, so there is a risk of bugs here.

why not make the AdyenContext.payment  private`var`  and in the `updated(with: payment)` just update the payment instead of creating a new instance :-) (AyenContext is a class)

Reply:
`DropInComponent.context` is holding an original payment and `ComponentManager.context` - the one is in use.
I'd like to keep `DropInComponent.context` untouched because if you opt-out from GiftCard - you need to get back to an original `payment`